addons/hr_timesheet/controllers/portal.py

```diff
@@ -135,17 +135,24 @@ def get_timesheets():
                     raw_timesheets_group = Timesheet_sudo._read_group(
                         domain, ['date:day'], ['unit_amount:sum', 'id:recordset'], order='date:day desc'
                     )
-                    grouped_timesheets = [(records, unit_amount) for __, unit_amount, records in raw_timesheets_group]
+                    grouped_timesheets = [(records, unit_amount, 0) for __, unit_amount, records in raw_timesheets_group]
 
                 else:
                     time_data = Timesheet_sudo._read_group(domain, [field], ['unit_amount:sum'])
-                    mapped_time = {field.id: unit_amount for field, unit_amount in time_data}
-                    grouped_timesheets = [(Timesheet_sudo.concat(*g), mapped_time[k.id]) for k, g in groupbyelem(timesheets, itemgetter(field))]
+                    mapped_time = {
+                        field.id: (unit_amount, field.allocated_hours if groupby in ['project_id', 'task_id', 'parent_task_id'] else 0)
```

Comment:
you're not verifying if the feature is enabled or not, i can set allocated_hours to > 0 and deactivate allow_timesheets (the value remains) and it will be displayed (you can test the flow)

![image](https://github.com/user-attachments/assets/e7e1a592-887e-4058-981e-adbee8190949)


```suggestion
                        field.id: (unit_amount, field.allocated_hours if groupby in ['project_id', 'task_id', 'parent_task_id'] and field.allow_timesheets else 0)
```

i also suggest separating the value and the possibility to display it,

if allocated_hours = 300 and i assign 0 to it because the feature is disabled, i'm not a big fan of similar approches, i would prefer having 2 fields a bool should_display_allocated_hours with is False for line 138, line 143 if allow_timesheet is false and line 155.

i will apply my suggestion to let you see it

Reply:
> you're not verifying if the feature is enabled or not, i can set allocated_hours to > 0 and deactivate allow_timesheets (the value remains) and it will be displayed (you can test the flow)
> 
> ![image](https://private-user-images.githubusercontent.com/64937092/386679824-e7e1a592-887e-4058-981e-adbee8190949.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5MTE3OTksIm5iZiI6MTczMTkxMTQ5OSwicGF0aCI6Ii82NDkzNzA5Mi8zODY2Nzk4MjQtZTdlMWE1OTItODg3ZS00MDU4LTk4MWUtYWRiZWU4MTkwOTQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE4VDA2MzEzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAxN2IyMDFiNWIyMTE5ODQwYTQ5MzJkYjJiMDc5Yjc0OGM5ZjMxMzU3ZDNiOTkzNWY5Y2EwNzE5MjIwODBmOWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bq3JhDpT6kVSyWOg6JZ91wYFaCtB9OFWPgmSMfwMhdA)
> 
> i also suggest separating the value and the possibility to display it,
> 
> if allocated_hours = 300 and i assign 0 to it because the feature is disabled, i'm not a big fan of similar approches, i would prefer having 2 fields a bool should_display_allocated_hours with is False for line 138, line 143 if allow_timesheet is false and line 155.
> 
> i will apply my suggestion to let you see it

Oh, I missed out. cc @diag-odoo